config/config_test.go

```diff
@@ -180,6 +181,92 @@ func TestInitConfig(t *testing.T) {
 	assert.Equal(t, "", param.Federation_DiscoveryUrl.GetString())
 }
 
+// Helper function to set and reset the environment variable HOME/USERPROFILE
+func setHomeDirEnv(t *testing.T, mockHomeDir string) func() {
+	// Save the original environment variables for restoration
+	originalHome := os.Getenv("HOME")
+
+	// Set the appropriate environment variable based on the platform
+	if err := os.Setenv("HOME", mockHomeDir); err != nil {
+		t.Fatalf("Failed to set HOME: %v", err)
+	}
+
+	// Return a function to restore the environment variables
+	return func() {
+		// Restore HOME
+		if originalHome != "" {
+			os.Setenv("HOME", originalHome)
+		} else {
+			os.Unsetenv("HOME")
+		}
+	}
+}
+
+func TestHomeDir(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		mockHomeDir := filepath.Join("test", "configDir")
+		ResetConfig()
+		t.Cleanup(func() {
+			ResetConfig()
+		})
+
+		// Save the original environment variables
+		oldConfigRoot := isRootExec
+		resetEnv := setHomeDirEnv(t, mockHomeDir)
+
+		defer func() {
+			resetEnv()
+			isRootExec = oldConfigRoot
+		}()
+
+		t.Run("RootUserNoConfigDir", func(t *testing.T) {
```

Comment:
Since each of these test cases share a very similar structure, I think they can be cleaned up by iterating over a slice of testing cases. Something along the lines of:
```
    type testCase struct {
        name         string
        isRootExec   bool
        configDir    string
        homeEnv      string
        expectedDir  string
    }

    testCases := []testCase{
        {
            name:        "RootUserNoConfigDir",
            isRootExec:  true,
            configDir:   "",
            homeEnv:     "",
            expectedDir: "/etc/pelican",
        },
        {
            name:        "NonRootWithConfigDir",
            isRootExec:  false,
            configDir:   "/test/configDir",
            homeEnv:     "",
            expectedDir: "/test/configDir",
        },
        ... (rest of test cases)
    }

    for _, tc := range testCases {
        t.Run(tc.name, func(t *testing.T) {
            isRootExec = tc.isRootExec
            viper.Reset()

            if tc.configDir != "" {
                viper.Set("ConfigDir", tc.configDir)
            }

            if tc.homeEnv != "" {
                os.Setenv("HOME", tc.homeEnv)
            } else {
                os.Unsetenv("HOME")
            }

            InitConfigDir(viper.GetViper())

            cDir := viper.GetString("ConfigDir")
            require.Equal(t, tc.expectedDir, cDir)
        })
    }
}
```

We use this structure elsewhere, and I think it's really useful to highlight what each individual test is checking for.